### ISSUE TITLE
Make TeamCity ConfigurationProperties Integration Opt-in Case-sensitive

### ISSUE THREAD

**Issue by olo-ntaylor** [nuke-build/nuke#921]
### Description

Hello, first issue I am logging here! Please forgive me if I don't get it quite right.

We have a situation where, for unfortunate reasons, we have a project that inherits two parameters: `GithubApiKey` and `GitHubApiKey` (note the capital/lowercase 'h'). Obviously, it would be ideal _not_ to be in this situation, but given the scope and risk of those changes, that's not an option for us right now.

This becomes an issue [here](https://github.com/nuke-build/nuke/blob/develop/source/Nuke.Common/CI/TeamCity/TeamCity.cs#L39), where the TeamCity configuration parameters are loaded into a dictionary, but they are loaded into a dictionary in a _case-insensitive_ manner, which causes Nuke to blow up with an exception: `An item with the same key has already been added. Key: GithubApiToken`.

### TL;DR

The new feature should allow TeamCity configuration parameters to be loaded into the `ConfigurationProperties` dictionary in a case-sensitive **or** case-insensitive manner, whichever way the Nuke consumer would prefer.

### Usage Example

```code
I imagine this opt-in could either be driven by...

* An environment variable, e.g. `NUKE_TEAMCITY_CONFIG_PARAMETERS_CASE_SENSITIVE=true`, or
* A public setter method on `TeamCity.Instance`, e.g. `.SetPropertiesAreCaseSensitive(true)`, or
* Something else! I'm honestly not super familiar (yet) with all the awesome config options available
```

### Alternative

There is no workaround at this point (besides _not_ having the ridiculous problem of having two distinct github api key parameters 😭).

I would be more than happy to help draft the PR for these changes. I just am not sure what is the ideal/recommended way to implement this feature yet!

**Comment by matkoch** [comment]
I'm adding a try-catch that returns `null` for the dictionary instead.
